From: Prarit Bhargava <prarit@redhat.com>
Subject: [RHEL 5.1 PATCH]: Remove PCIE warning for devices with no irq pin
Date: Fri, 23 Mar 2007 11:50:55 -0400
Bugzilla: 219318
Message-Id: <20070323155055.10113.30035.sendpatchset@prarit.boston.redhat.com>
Changelog: [pcie]: Remove PCIE warning for devices with no irq pin


Backport of

http://kernel.org/pub/linux/kernel/people/akpm/patches/2.6/2.6.19/2.6.19-mm1/dont-use/broken-out/gregkh-pci-pci-pcieport-driver-remove-invalid-warning-message.patch

which removes erroneous warning

	pcie_portdrv_probe->Dev[XXXX:XXXX] has invalid IRQ. Check vendor BIOS

for devices that do not have an interrupt pin.

Compile & boot tested by me.  (I couldn't find a system that had this problem.
The patch seems to have resolved the issue upstream, and is a one-liner to
the PCIE Port Driver.  I'm comfortable with the patch since it is low-risk...)

Resolves bz219318.

P.

--- linux-2.6.18.ia64.orig/drivers/pci/pcie/portdrv_pci.c	2007-03-23 11:41:06.000000000 -0400
+++ linux-2.6.18.ia64/drivers/pci/pcie/portdrv_pci.c	2007-03-23 11:43:10.000000000 -0400
@@ -51,7 +51,7 @@ static int __devinit pcie_portdrv_probe 
 		return -ENODEV;
 	
 	pci_set_master(dev);
-        if (!dev->irq) {
+        if (!dev->irq && dev->pin) {
 		printk(KERN_WARNING 
 		"%s->Dev[%04x:%04x] has invalid IRQ. Check vendor BIOS\n", 
 		__FUNCTION__, dev->device, dev->vendor);

